client/sound/index.jsx

```diff
@@ -14,67 +15,65 @@ export default class Sound extends Component {
   constructor (props) {
     super(props)
 
-    window.localStorage.setItem('sound-window', true)
-
-    window.onbeforeunload = function () {
-      return 'Closing this window will stop the sounds of the timer. Are you sure you want to close it?'
-    }
-
-    window.onunload = () => {
-      window.localStorage.removeItem('sound-window')
-    }
-
-    window.onstorage = event => {
-      if (event.key === 'focus' && event.newValue) {
-        window.focus()
-        window.localStorage.removeItem('focus')
-      }
-    }
-
     onStartEvent(() => {
-      this.testSound(startSound)
+      this.soundEvent(startSound)
     })
 
     onEndGameEvent(() => {
-      this.testSound(endgameSound)
+      this.soundEvent(endgameSound)
     })
 
     onEndEvent(() => {
-      this.testSound(endSound)
+      this.soundEvent(endSound)
     })
 
     onStopEvent(() => {
-      this.testSound(stopSound)
+      this.soundEvent(stopSound)
     })
 
-    this.state = {}
+    this.soundEvent = sound => this.playSound(sound)
+    this.state = { enabled: true }
   }
 
-  testSound (sound) {
+  playSound (sound) {
     const audio = new window.Audio(sound)
     audio.play()
       .catch(err => {
         console.error(err)
       })
   }
 
+  toggleState (isEnabled) {
+    if (isEnabled) {
+      this.setState({ enabled: false })
+      this.soundEvent = () => {}
```

Comment:
Wouldn't it'd be easier to check the state in the function instead of changing the function every time? That way you also wouldn't need the `soundEvent` variable at all.